examples_apps/run_state_learner.py

```diff
@@ -0,0 +1,432 @@
+   optimization. The TensorFlow backend will perform numerical
+   operations in this truncated Fock space when performing the
+   optimization.
+
+-  ``depth``: The number of layer ansatz in our variational quantum
+   circuit. As a general rule, increasing the number of layers (and
+   thus, the number of parameters we are optimizing over) increases the
+   optimizers chance of finding a reasonable local minimum in the
+   optimization landscape.
+
+-  ``reps``: the number of steps in the optimization routine performing
+   gradient descent
+
+Some other optional hyperparameters include:
+
+-  The standard deviation of initial parameters. Note that we make a
+   distinction between the standard deviation of *passive* parameters
+   (those that preserve photon number when changed, such as phase
+   parameters), and *active* parameters (those that introduce or remove
+   energy from the system when changed).
+"""
+
+import strawberryfields as sf
+from strawberryfields.ops import *
+from strawberryfields.utils import operation
+
+# Cutoff dimension
+cutoff = 9
+
+# Number of layers
+depth = 15
+
+# Number of steps in optimization routine performing gradient descent
+reps = 200
+
+# Learning rate
+lr = 0.05
+
+# Standard deviation of initial parameters
+passive_sd = 0.1
+active_sd = 0.001
+
+######################################################################
+# The layer parameters :math:`\vec{\theta}`
+# -----------------------------------------
+#
+# We use TensorFlow to create the variables corresponding to the gate
+# parameters. Note that each variable has shape ``[depth]``, with each
+# individual element representing the gate parameter in layer :math:`i`.
+
+import tensorflow as tf
+
+# set the random seed
+tf.random.set_seed(42)
+
+# squeeze gate
+sq_r = tf.random.normal(shape=[depth], stddev=active_sd)
+sq_phi = tf.random.normal(shape=[depth], stddev=passive_sd)
+
+# displacement gate
+d_r = tf.random.normal(shape=[depth], stddev=active_sd)
+d_phi = tf.random.normal(shape=[depth], stddev=passive_sd)
+
+# rotation gates
+r1 = tf.random.normal(shape=[depth], stddev=passive_sd)
+r2 = tf.random.normal(shape=[depth], stddev=passive_sd)
+
+# kerr gate
+kappa = tf.random.normal(shape=[depth], stddev=active_sd)
+
+
+######################################################################
+# For convenience, we store the TensorFlow variables representing the
+# weights as a tensor, as well as creating the corresponding
+# single-mode Strawberry Fields program and parameters:
+
+weights = tf.convert_to_tensor([r1, sq_r, sq_phi, r2, d_r, d_phi, kappa])
+weights = tf.Variable(tf.transpose(weights))
+
+# Single-mode Strawberry Fields program
+prog = sf.Program(1)
+
+# Create the Strawberry Fields free parameters for each layer
+sf_params = []
+names = ["r1", "sq_r", "sq_phi", "r2", "d_r", "d_phi", "kappa"]
+
+for k in range(depth):
+    sf_params_names = ["{}_{}".format(n, k) for n in names]
+    sf_params.append(prog.params(*sf_params_names))
```

Comment:
(very minor)
```suggestion
for i in range(depth):
    sf_params_names = ["{}_{}".format(n, i) for n in names]
    sf_params.append(prog.params(*sf_params_names))
```
Since in the next line we talk about the i-th layer.